Ipython version detection update and new drivers

ggdeboo opened the pull request:
I have modified misc.py to detect the version of newer ipython releases and added a bunch of instrument wrappers.

Commit:
Committer: Gabriele de Boo <ggdeboo@gmail.com>

Fix for the detection of newer versions of ipython.

ggdeboo committed:
New instrument wrappers.

ggdeboo committed:
Fixed the unpacking of arguments in functionframe.py.

heeres commented on source/lib/misc.py:
```diff
@@ -25,12 +26,6 @@ def get_dict_keys(dic, keys):
-    for key, val in dic.iteritems():
-        ret[val] = key
-    return ret
-
```
Is there a reason for dropping this?

ggdeboo replied:
It doesn't seem to be used anywhere.

heeres commented on source/lib/misc.py:
```diff
@@ -100,35 +95,39 @@ def usleep(usec):
     while (exact_time() - start) * 1e6 < usec:
         pass
 
```
It would be good to drop instead of comment code. History will be in git anyway.

However, I am a little worried about backwards-compatibility here. I think these changes will make anything older than IPython 1.0 not work. Then again, that might not matter too much as that release was somewhere in 2013. We might want to put it in the README or something then.

ggdeboo replied:
I'll have a go at making it backwards compatible. What do you think is the lowest version of Python that should be supported?

9x replied:
I don't think there is a good reason to use an outdated version of IPython, but if you are concerned about it, we could just add another case instead of directly importing get_ipython. But I am not sure in which version of IPython this was changed
```python
    if ipython_is_newer((3,0,0)):
        return IPython.get_ipython()
```

heeres replied:
I think I agree it's ok to drop compatibility for <1.0.

get_python was introduced here:
https://github.com/ipython/ipython/commit/790f7312fb7f317b5f2afcbf7abb41dfa59c3610

Which is at least in the 1.0 release.

So perhaps you could drop all the commented code that is not useful anymore
and I'll merge it.

Thanks and regards,

On Tue, Apr 14, 2015 at 9:50 PM, Jens Mohrmann <notifications@github.com>
wrote:

> In source/lib/misc.py
> <https://github.com/heeres/qtlab/pull/3#discussion_r28388142>:
>
> > @@ -100,35 +95,39 @@ def usleep(usec):
> >      while (exact_time() - start) * 1e6 < usec:
> >          pass
> >
>
> I don't think there is a good reason to use an outdated version of
> IPython, but if you are concerned about it, we could just add another case
> instead of directly importing get_ipython. But I am not sure in which
> version of IPython this was changed
>
>     if ipython_is_newer((3,0,0)):
>         return IPython.get_ipython()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/heeres/qtlab/pull/3/files#r28388142>.
>



-- 
Reinier Heeres
Tel: +1 203 745 7955

ggdeboo committed:
Removed unnecessary comments from misc.py.
Added requirements for IPython and pyVISA to make it clear which versions are
not supported.

ggdeboo committed:
Update for the requirements in the INSTALL file

ggdeboo commented:
I have removed the commented code. Is there anything else that should be done before this is merged?

heeres commented:
thanks!
